NightscoutUploadKit/NightscoutUploader.swift

```diff
@@ -57,7 +58,7 @@ public class NightscoutUploader {
         lastStoredTreatmentTimestamp = nil
     }
 
-    public init(siteURL: String?, APISecret: String?) {
+    public init(siteURL: NSURL?, APISecret: String?) {
```

Comment:
Now that this is wrapped in the `NightscoutService` struct, which only creates it when these two arguments are non-nil, you could simplify here and make these non-optional